ietf/meeting/models.py

```diff
@@ -244,18 +244,31 @@ def get_attendance(self):
         number = self.get_number()
         if number is None or number < 110:
             return None
-        Attendance = namedtuple('Attendance', 'onsite online')
+        Attendance = namedtuple('Attendance', 'onsite remote')
+
+        # MeetingRegistration.attended started conflating badge-pickup and session attendance before IETF 114.
+        # We've separated session attendence off to ietf.meeting.Attended, but need to report attendance at older
+        # meetings correctly.
+
+        attended_per_meetingregistration = (
+            Q(meetingregistration__meeting=self) & (
+                Q(meetingregistration__attended=True) |
+                Q(meetingregistration__checkedin=True)
+            )
+        )
+        attended_per_meeting_attended = (
+            Q(attended__session__meeting=self)
+            # Note that we are not filtering to plenary, wg, or rg sessions
+            # as we do for nomcom eligibility - if picking up a badge (see above)
+            # is good enough, just attending e.g. a training session is also good enough
+        )
+        attended = Person.objects.filter(
+            attended_per_meetingregistration | attended_per_meeting_attended
+        ).distinct()
```

Comment:
There seem to be people who have attended sessions but have not registered. E.g.,
```
(Pdb) attended.filter(meetingregistration__meeting=self).count()
1290
(Pdb) attended.exclude(meetingregistration__meeting=self).count()
11
```

Reply:
Yes, this is expected - Note the comment about attending e.g. a training session. Could also have been the hackathon.

```
    attended.exclude(meetingregistration__meeting=self).values_list("attended__session__group__acronym"): '<QuerySet [('hackathon',), ('can',)]>'
```

Should we explicitly exclude hackathon? If so, what else should be excluded. Do we revert to only counting wg, rg, and plenary sessions?

Reply:
Ah, I see the significance of the comment now. As it stands, I think these records will be ignored - without a meetingregistration, they're ignored by the queries for onsite and remote registrations. Maybe that's ok?

Reply:
Yes, the people who didn't register as onsite or remote won't be counted. To reconvince myself that made it all ok, I reconsidered the case that someone registered to be remote, but only ever showed up for the hackathon and questioned whether they should be counted. I landed on "yes" because someone could register onsite, pick up their badge, and only go to the hackathon and they would get counted.